editor/plugins/text_shader_editor.cpp

```diff
@@ -268,17 +268,35 @@ void ShaderTextEditor::_load_theme_settings() {
 				}
 			}
 
-			const Vector<ShaderLanguage::ModeInfo> &modes = ShaderTypes::get_singleton()->get_modes(RenderingServer::ShaderMode(i));
+			{
+				const Vector<ShaderLanguage::ModeInfo> &render_modes = ShaderTypes::get_singleton()->get_modes(RenderingServer::ShaderMode(i));
+
+				for (int j = 0; j < render_modes.size(); j++) {
+					const ShaderLanguage::ModeInfo &mode_info = render_modes[j];
+
+					if (!mode_info.options.is_empty()) {
+						for (int k = 0; k < mode_info.options.size(); k++) {
+							built_ins.push_back(String(mode_info.name) + "_" + String(mode_info.options[k]));
```

Comment:
```suggestion
				for (const ShaderLanguage::ModeInfo &mode_info : render_modes) {

					if (!mode_info.options.is_empty()) {
						for (const StringName &option : mode_info.options) {
							built_ins.push_back(String(mode_info.name) + "_" + String(option));
```
Use for-each syntax when applicable